experiments/kompanion/cmd/lint/lint.go

```diff
@@ -0,0 +1,288 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lint
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"sort"
+	"strings"
+	"sync"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/experiments/kompanion/pkg/utils"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	examples = `
+	# lint Config Connector resources across all namespaces, excludes \"kube\" namespaces by default
+	kompanion lint
+
+	# exclude certain namespace prefixes
+	kompanion lint --exclude-namespaces=kube --exclude-namespaces=my-team
+
+	# target only specific namespace prefixes
+	kompanion lint --target-namespaces=my-team
+
+	# include more resource kind in linting
+	kompanion lint --include-resources=ComputeNetwork
+
+	# ignore more resource kind in linting
+	kompanion lint --ignore-resources=StorageBucket
+	`
+)
+
+var abandonOnDeleteCheckKind = map[string]bool{
+	"AlloyDBCluster":    true,
+	"AlloyDBInstance":   true,
+	"BigQueryDataset":   true,
+	"BigQueryTable":     true,
+	"BigtableInstance":  true,
+	"ContainerCluster":  true,
+	"ContainerNodePool": true,
+	"KMKeyRing":         true,
+	"KMSCryptoKey":      true,
+	"RedisCluster":      true,
+	"SpannerDatabase":   true,
+	"SpannerInstance":   true,
+	"SQLDatabase":       true,
+	"SQLInstance":       true,
+	"StorageBucket":     true,
+}
+
+const (
+	kubeconfigFlag = "kubeconfig"
+
+	targetNamespacesFlag = "target-namespaces"
+	ignoreNamespacesFlag = "exclude-namespaces"
```

Comment:
NOT for this PR. It would be nice to have a mode for namespace mode which only worked with namespaces having a valid CCC.